### PR TITLE
add ignore reviewers change to alwaysIgnoreNonCodeRelatedUpdates

### PR DESCRIPTION
**Pull request by mifitous**
In case that reviewers list changed, but flag alwaysIgnoreNonCodeRelatedUpdates was checked, ignore build

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by jetersen — APPROVED** [review]

**Comment by jetersen** [comment]
@mifitous https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fgitlab-branch-source-plugin/detail/PR-295/1/pipeline/43#step-66-log-120

`getReviewers` do not exist.

**Comment by mifitous** [comment]
> @mifitous https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fgitlab-branch-source-plugin/detail/PR-295/1/pipeline/43#step-66-log-120
> 
> `getReviewers` do not exist.

Hi @jetersen 
It was merged in https://github.com/gitlab4j/gitlab4j-api/pull/916 but I do not understand why it is still not presented in our code. Any idea?

**Comment by mifitous** [comment]
> @mifitous
> 
> https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/ad1ce6d01959377bf3ff35f1905b5ce941e501d5/pom.xml#L36-L39
> 
> https://github.com/jenkinsci/bom/blob/bd41cba72c8231b42ad4fc355a42d92e14fad96f/bom-weekly/pom.xml#L139-L141
> 
> https://github.com/jenkinsci/gitlab-api-plugin/blob/3b564bafb14281a9e0e85367c4473eafedfb6f82/pom.xml#L77-L79
> 
> https://github.com/jenkinsci/gitlab-api-plugin/blob/3b564bafb14281a9e0e85367c4473eafedfb6f82/pom.xml#L17

so I'll wait for it...
Thanks

**Comment by jetersen** [comment]
Feel free to submit a PR to update gitlab4j in the gitlab-api-plugin. I don't think Dependabot have caught it yet.

**Comment by jetersen** [comment]
https://github.com/jenkinsci/gitlab-api-plugin/pull/41 FYI

**Comment by jetersen** [comment]
@mifitous You can rebase the PR on default branch now to get the updated GitLab API :)

**Comment by mifitous** [comment]
> @mifitous You can rebase the PR on default branch now to get the updated GitLab API :) #298

Hi @jetersen 
Merged, but still not passing. Seems https://github.com/gitlab4j/gitlab4j-api/pull/916 was merged but the gitlab4j-api was not released with this MR. https://github.com/gitlab4j/gitlab4j-api/releases

**Comment by jetersen** [comment]
@mifitous I got to the same conclusion 😓

Let's hope we can get a release with the changes :)

**Comment by jmini** [comment]
The change is available in `5.2.0`:

I think you need https://github.com/jenkinsci/gitlab-api-plugin/pull/44

**Comment by jmini** [comment]
I think the approach of checking the `changes` value for the `alwaysIgnoreNonCodeRelatedUpdates` ("Ignore builds for non-code related MR updates") feature is not good enough.

As I wrote in https://github.com/jenkinsci/gitlab-branch-source-plugin/issues/323 the presence or the absence of `object_attributes.oldrev` seems to be the correct approach to cover all cases.

**Comment by mifitous** [comment]
> @mifitous I got to the same conclusion 😓
> 
> Let's hope we can get a release with the changes :)

Hi @jetersen : indeed, now it compile fine with the new version of the pom.xml. all tests passed

File: pom.xml [hunk 1]
```diff
@@ -40,8 +40,8 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.361.x</artifactId>
-        <version>2025.v816d28f1e04f</version>
+        <artifactId>bom-2.387.x</artifactId>
+        <version>2179.v0884e842b_859</version>
```
Comment (jetersen) [inline pom.xml:44]: Can we do this in a separate PR?

Or is it asking too much ? 😓

Comment (mifitous) [reply pom.xml:44]: Hi @jetersen 
sure: https://github.com/jenkinsci/gitlab-branch-source-plugin/pull/334

**Comment by mifitous** [comment]
> @mifitous what do you think @jmini comment above
> 
> > I think the approach of checking the `changes` value for the `alwaysIgnoreNonCodeRelatedUpdates` ("Ignore builds for non-code related MR updates") feature is not good enough.
> > As I wrote in #323 the presence or the absence of `object_attributes.oldrev` seems to be the correct approach to cover all cases.

Hi @jetersen and @jmini 

as https://github.com/gitlab4j/gitlab4j-api/pull/980 is now merged, I'll go for implementation of it
I'll correct this PR

**Comment by jetersen** [comment]
@mifitous Much appreciated and sorry for the rework 😅

**Comment by mifitous** [comment]
Created new PR at: https://github.com/jenkinsci/gitlab-branch-source-plugin/pull/335
